urlwatch

```diff
@@ -94,304 +61,51 @@ except AttributeError:
-            job_state.save()
-
-    # Output everything
-    report.finish()
-
-    # Close cache
-    cache_storage.close()
-
-
 if __name__ == '__main__':
-    parser = argparse.ArgumentParser(description=urlwatch.__doc__,
-                                     formatter_class=argparse.RawDescriptionHelpFormatter)
 
-    parser.add_argument('--version', action='version', version='%(prog)s {}'.format(urlwatch.__version__))
-    parser.add_argument('-v', '--verbose', action='store_true', help='show debug output')
+    config_file = os.path.join(urlwatch_dir, CONFIG_FILE)
+    urls_file = os.path.join(urlwatch_dir, URLS_FILE)
+    cache_file = os.path.join(urlwatch_dir, CACHE_FILE)
+    hooks_file = os.path.join(urlwatch_dir, HOOKS_FILE)
+
+    command_config = CommandConfig(pkgname, urlwatch_dir, bindir, prefix, config_file, urls_file, hooks_file, cache_file, False)
+    setup_logger(command_config.verbose)
+    # command_config.migrate_cache = False
 
-    group = parser.add_argument_group('files and directories')
-    group.add_argument('--urls', metavar='FILE', help='read job list (URLs) from FILE', default=urls_yaml)
-    group.add_argument('--config', metavar='FILE', help='read configuration from FILE', default=config_yaml)
-    group.add_argument('--hooks', metavar='FILE', help='use FILE as hooks.py module', default=hooks_py)
-    group.add_argument('--cache', metavar='FILE', help='use FILE as cache database', default=cache_db)
+    # setup storage API
+    config_storage = YamlConfigStorage(command_config.config)
+    # config_storage = JsonConfigStorage(command_config.config)
+    cache_storage = CacheMiniDBStorage(command_config.cache)
+    # cache_storage = CacheDirStorage(command_config.cache)
+    # urls_storage = UrlsJson(command_config.urls)
+    urls_storage = UrlsYaml(command_config.urls)
 
-    group = parser.add_argument_group('job list management')
-    group.add_argument('--list', action='store_true', help='list jobs')
-    group.add_argument('--add', metavar='JOB', help='add job (key1=value1,key2=value2,...)')
-    group.add_argument('--delete', metavar='JOB', help='delete job by location or index')
+    urlwatcher = Urlwatcher(command_config, config_storage, cache_storage, urls_storage)
 
-    group = parser.add_argument_group('interactive commands ($EDITOR/$VISUAL)')
-    group.add_argument('--edit', action='store_true', help='edit URL/job listL')
-    group.add_argument('--edit-config', action='store_true', help='edit configuration file')
-    group.add_argument('--edit-hooks', action='store_true', help='edit hooks script')
+    urlwatch_command = UrlwatchCommand(urlwatcher)
+    urlwatch_command.run()
 
-    group = parser.add_argument_group('miscellaneous')
-    group.add_argument('--features', action='store_true', help='list supported jobs/filters/reporters')
-    group.add_argument('--gc-cache', action='store_true', help='remove old cache entries')
+    # urlwatch_config = UrlwatchConfig(pkgname, bindir, prefix)
```

Comment:
Do not leave commented-out code here. Either use it or remove it. If we need it in the future, we can always dig it out from the Git history.